### PR TITLE
Vend::Tax and Tax Jar API

### PR DESCRIPTION
**Pull request by msjohns1**
* Vend::Tax - a general-purpose linking module
  between Interchange and specific 3rd-party tax
  APIs.

* Vend::Tax::TaxJar - interface to Tax Jar for
  live tax lookups.

* Integrate support for Vend::Tax into strap demo.

* Fix tax recomputation in admin to work when 3rd-
  party API tax in effect.

* Per testing, fixed broken stock_alert dbconf for
  SQLite.

### CODE REVIEW

File: dist/strap/dbconf/mysql/transactions.mysql [hunk 1]
```diff
@@ -57,3 +57,4 @@ Database  transactions  COLUMN_DEF   "comments=text"
 Database  transactions  COLUMN_DEF   "currency_locale=varchar(32)"
 Database  transactions  COLUMN_DEF   "pay_cert_total=varchar(16)"
 Database  transactions  COLUMN_DEF   "pay_cert_ord_total=varchar(16)"
+Database  transactions  COLUMN_DEF   "tax_sent=enum('1','-1') not null default '1'"
```
Comment (jonjensen) [inline dist/strap/dbconf/mysql/transactions.mysql]: As we discussed, I would love to see some documentation here of the weird MySQL enum behavior that leads to not having an explicit `'0'` value. Possibly including a link to the [MySQL enum docs](https://dev.mysql.com/doc/refman/5.7/en/enum.html#enum-nulls).

File: dist/strap/include/checkout/tax_popup [hunk 2]
```diff
@@ -1,4 +1,7 @@
-[if config SalesTax ne multi]
+[if type=!explicit compare="
+    local $_ = $Config->{SalesTax};
+    return $_ eq 'multi' || /^\[/;
```
Comment (jonjensen) [inline dist/strap/include/checkout/tax_popup]: Is this now treating SalesTax set to an ITL tag similarly to `multi`? Just trying to understand this change.

**Comment by msjohns1** [comment]
>> @@ -1,4 +1,7 @@
> -[if config SalesTax ne multi]
> +[if type=!explicit compare="
> +    local $_ = $Config->{SalesTax};
> +    return $_ eq 'multi' || /^\[/;
> 
> Is this now treating SalesTax set to an ITL tag similarly to `multi`? Just trying to understand this change.

Yes, that is correct. It suppresses the calculation popup when SalesTax
is either "multi" or set to ITL. I think the actual intent of the popup
is to run when SalesTax is either "state" or "zip", but I can't be sure.

If the concern is whether the popup is viable potentially for other ITL
used in SalesTax, I can expand that regex to '\[tax[-_]lookup' instead.

**Comment by jonjensen** [comment]
> If the concern is whether the popup is viable potentially for other ITL used in SalesTax, I can expand that regex to '\[tax[-_]lookup' instead.

I did wonder about that. Matching the exact tag like you suggested seems wise.

File: dist/strap/dbconf/mysql/transactions.mysql (hunk 1 above)
Comment (msjohns1) [reply dist/strap/dbconf/mysql/transactions.mysql]: Change made and pushed.

File: dist/strap/include/checkout/tax_popup (hunk 2 above)
Comment (msjohns1) [reply dist/strap/include/checkout/tax_popup]: 2nd condition expanded out to match on `/^\[tax[-_]lookup/` instead.

**Comment by jonjensen** [comment]
@machack666 Ok, any yea or nay from you?

**Pull request merged**